src/components/containers/MoviesContainer.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { Movie, Page, SortBy } from "../../models";
+import MoviesResultPane from "../renders/MoviesResultPane";
+import ResultSummaryBar from "../renders/ResultSummaryBar";
+import { sortedBy } from "./options.json";
+
+function map(state: Page): { movies: Movie[] } {
+  const { movies } = state;
+  return { movies: movies };
+}
+
+function getComparator(sortBy: SortBy): (one: Movie, two: Movie) => number {
+  switch (sortBy) {
+    case SortBy.RATING:
+      return (one, two): number => one.voteAverage - two.voteAverage;
+    case SortBy.REALIZE_DATE:
+      return (one, two): number =>
+        one.releaseDate.getTime() - two.releaseDate.getTime();
+    default:
+      return (one, two): number => two.id - one.id;
+  }
+}
+
+class MoviesContainer extends React.Component<
+  { movies: Movie[] },
+  { sortBy: SortBy }
+> {
+  constructor(props: { movies: Movie[] }) {
+    super(props);
+    this.state = { sortBy: SortBy.REALIZE_DATE };
+    this.onClickToChoose = this.onClickToChoose.bind(this);
```

Comment:
If you don't want to use bindings in the constructor you can rewrite it using this approach:
```
class MoviewContainer {
   ...
   onClickToChoose = () => {
      console.log(this);
    }
}
```